### PR TITLE
Refined some details.

### PR DESCRIPTION
**Pull request by sundargs2000**
Changes made -
1. Services which will not be routed will be created before the buffer time.
2. Cleaner buffer time log message.
3. Removed cleanup function used during reject.
4. Updated readme

### CODE REVIEW

File: src/utilities/strategy-helpers/blue-green-helper.ts [hunk 1]
```diff
@@ -167,14 +146,25 @@ export function getManifestObjects (filePaths: string[]): any {
...
     serviceEntityList.forEach(inputObject => {
         const name = inputObject.metadata.name;
         serviceNameMap.set(name, getBlueGreenResourceName(name, GREEN_SUFFIX));
     });
      
-    return { serviceEntityList: serviceEntityList, serviceNameMap: serviceNameMap, deploymentEntityList: deploymentEntityList, ingressEntityList: ingressEntityList, otherObjects: otherEntitiesList };
+    return { serviceEntityList: serviceEntityList, serviceNameMap: serviceNameMap, unroutedServiceEntityList: unroutedServiceEntityList, deploymentEntityList: deploymentEntityList, ingressEntityList: ingressEntityList, otherObjects: otherEntitiesList };
 }
 
 export function isServiceRouted(serviceObject: any[], deploymentEntityList: any[]): boolean {
```
Comment (ajinkya599) [inline src/utilities/strategy-helpers/blue-green-helper.ts:170]: Is `UnsetsClusterSpecficDetails` being used anywhere?

Comment (sundargs2000) [reply src/utilities/strategy-helpers/blue-green-helper.ts:170]: yes in `fetchResource`

File: src/utilities/strategy-helpers/smi-blue-green-helper.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
 import { Kubectl } from '../../kubectl-object-model';
 import * as kubectlUtils from '../kubectl-util';
 import * as fileHelper from '../files-helper';
-import { createWorkloadsWithLabel, getManifestObjects, fetchResource, deleteWorkloadsWithLabel, cleanUp, getNewBlueGreenObject, getBlueGreenResourceName, isServiceRouted, deleteObjects } from './blue-green-helper';
+import { createWorkloadsWithLabel, getManifestObjects, fetchResource, deleteWorkloadsWithLabel, getNewBlueGreenObject, getBlueGreenResourceName, deleteObjects } from './blue-green-helper';
 import { GREEN_LABEL_VALUE, NONE_LABEL_VALUE, GREEN_SUFFIX, STABLE_SUFFIX } from './blue-green-helper';
 
 let trafficSplitAPIVersion = "";
```
[the remark this answers is not in the document]
Comment (sundargs2000) [reply src/utilities/strategy-helpers/smi-blue-green-helper.ts:9]: i am using it now

File: src/utilities/strategy-helpers/blue-green-helper.ts [hunk 3]
```diff
@@ -167,14 +146,25 @@ export function getManifestObjects (filePaths: string[]): any {
...
+
     let serviceNameMap = new Map<string, string>();
     // find all services and add their names with blue green suffix
     serviceEntityList.forEach(inputObject => {
         const name = inputObject.metadata.name;
         serviceNameMap.set(name, getBlueGreenResourceName(name, GREEN_SUFFIX));
     });
      
-    return { serviceEntityList: serviceEntityList, serviceNameMap: serviceNameMap, deploymentEntityList: deploymentEntityList, ingressEntityList: ingressEntityList, otherObjects: otherEntitiesList };
+    return { serviceEntityList: serviceEntityList, serviceNameMap: serviceNameMap, unroutedServiceEntityList: unroutedServiceEntityList, deploymentEntityList: deploymentEntityList, ingressEntityList: ingressEntityList, otherObjects: otherEntitiesList };
```
[the remark this answers is not in the document]
Comment (sundargs2000) [reply src/utilities/strategy-helpers/blue-green-helper.ts]: wouldnt it be inconsistent with `serviceEntityList` ?

**Comment by sundargs2000** [comment]
Need to add secrets L2_REPO_TOKEN.

File: src/utilities/strategy-helpers/blue-green-helper.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajinkya599) [reply src/utilities/strategy-helpers/blue-green-helper.ts]: Call it `routedServiceEntityList` as well

File: src/utilities/strategy-helpers/blue-green-helper.ts [hunk 4]
```diff
@@ -240,36 +228,30 @@ export function getBlueGreenResourceName(name: string, suffix: string) {
...
+    } else return null;
 }
 
-export function isServiceSelectorSubsetOfMatchLabel(serviceSelector: string, matchLabels: string): boolean {
+export function isServiceSelectorSubsetOfMatchLabel(serviceSelector: any, matchLabels: any): boolean {
     let serviceSelectorMap = new Map();
     let matchLabelsMap = new Map();
   
-    JSON.parse(serviceSelector, (key, value) => {
+    JSON.parse(JSON.stringify(serviceSelector), (key, value) => {
```
Comment (ajinkya599) [inline src/utilities/strategy-helpers/blue-green-helper.ts:250]: Check this(and below) once.

File: .github/workflows/intergation-tests.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+name: "Trigger Integration tests"
+on:
```
Comment (ajinkya599) [inline .github/workflows/intergation-tests.yml:2]: Fix typo in file name.

**Comment by ds-ms** [comment]
@ajinkya599 what do we do about this PR now?

**Review by ajinkya599 — APPROVED** [review]
🚀

**Comment by ds-ms** [comment]
@ajinkya599 why'd you merge it when the tests are failing?

**Pull request merged**